### PR TITLE
Fix tarball

### PR DESCRIPTION
**Pull request by ocefpaf**
B/c this project uses `setuptools_scm` the GitHub tarball in the release is not installable. It is not a GitHub repo and has no VCS info for the version.

This PR fixes some minor issues with the tarball, adds some extra checks, and upload the sdist tarball to GH artifact. I'm researching on how to move those artifacts to the GH release itself, not sure how to do that yet.

### CODE REVIEW

**Comment by ocefpaf** [comment]
The upload artifact only stores it on GH, we still need to attach that to the release assets. One can do that manually but ideally we should automate it. This GH action is promising https://github.com/marketplace/actions/upload-a-release-asset but I don't have the time to set it up here.

One example of the asset is in https://github.com/pangeo-data/xESMF/actions/runs/282797156. That zip file has an sdist tarball and a wheel. Both are tested and pip installable outside of the VCS.

**Comment by ocefpaf** [comment]
> Should we bump the version to 0.4.1 ?

This PR is only infrastructure for releases and, without a way to send those artifacts to GitHub release asset files, it won't do anything for the users.

If you want to test this though you can tag a 0.4.1, download the artifact, and re-upload them to the release file asset list. That part must be automated or no one will remember to do so every release.

**Comment by huard** [comment]
You lost me here, I'll merge as is then.

**Comment by ocefpaf** [comment]
Sorry. I was not clear.

There is no PyPI source distribution at the moment and the GitHub release/tag asset in [1] is a GitHub "frozen" state, not really. Source distributions are created with `python setup.py sdist` BTW.

The GitHub release brings two problems here:

- a minor one where the tarball is bloated with files that are not needed and;
- a big one where `setuptools_scm` expects to operate in a GitHub repsitory to extract the version.

If you try to download the current release and install it you'll get the error in [2]. If you create the source distribution with `python setup.py sdist` those issues are gone.

The changes in this PR creates the source distribution, test it t ensure it is installable, and upload as a GitHub artifact [3]. The problem is that the artifact we create with these new steps on the GH action goes to an obscure page [3] and not the release asset page [1].

One can upload it manually there or, what I recommend to do, we could implement [4] to perform that step automatically for us.

[1] https://github.com/pangeo-data/xESMF/releases/tag/v0.4.0
[2] https://gist.github.com/ocefpaf/755b904e53ec2762c402476604934036
[3] https://github.com/pangeo-data/xESMF/actions/runs/282797156
[4] https://github.com/marketplace/actions/upload-a-release-asset

PS: sure we can also ditch `setuptools_scm` and manually tag the version, that would also fix the problem here ;-p

**Comment by huard** [comment]
Got it, but I don't know how to upload the artifact on the release page... 

No attachment to `setuptools_scm`, if dropping it can simplify releases, I have no objection.

**Comment by ocefpaf** [comment]
When creating/editing a release you'll see this:

assets

You can attached the tarball and the wheel there. Just bare in mind that, if you want to use the artifacts created during the GH action phase, you'll have to unzip it first.

This approach is fine for now but not sustainable in the long term. Dropping `setuptools_scm` is a solution but at the same time it is nice to have tags attached to the version, one less thing to worry about then releasing.

TL;DR ideally we should automate the artifact to release asset upload and/or get control of the PyPI release again.
